cli/src/git_util.rs

```diff
@@ -172,17 +173,122 @@ pub fn with_remote_git_callbacks<T>(
     f(callbacks)
 }
 
-pub fn print_git_import_stats(ui: &mut Ui, stats: &GitImportStats) -> Result<(), CommandError> {
+pub fn print_git_import_stats(
+    ui: &mut Ui,
+    stats: &GitImportStats,
+    query_ref_stats: Option<&ReadonlyRepo>,
+) -> Result<(), CommandError> {
+    if let (Some(repo), true) = (query_ref_stats, !stats.changed_remote_refs.is_empty()) {
+        let mut remote_branches = Vec::new();
+        let mut tags = Vec::new();
+        let mut local_branches = Vec::new();
+
+        // group the references by type
+        for (refname, (remote_ref, ref_target)) in &stats.changed_remote_refs {
+            let ref_status = RefStatus::new(refname, remote_ref, ref_target, repo);
+            match refname {
+                RefName::RemoteBranch { .. } => remote_branches.push(ref_status),
+                RefName::Tag(_) => tags.push(ref_status),
+                RefName::LocalBranch(_) => local_branches.push(ref_status),
+            }
+        }
+
+        let grouped_refs = vec![remote_branches, tags, local_branches];
```

Comment:
nit: no need to split the changes into groups since the list is sorted anyway.

Reply:
resolved